CONTRIBUTING.md

```diff
@@ -167,16 +176,6 @@ Check out the [official TypeScript documentation](https://www.typescriptlang.org
 
 _Note that we have separate tests for our TypeScript definition files. We only use `ts-check` for local development and don't check it anywhere else like on the CI._
 
-### Why does the code base often use `let` instead of `const`?
-
-There is no real reason for that other a historical one. Back before auto-formatting via prettier was a thing and minifiers weren't as advanced as they are today we used a pretty terse code-style. The code-style deliberately was aimed at making code look as concise and short as possible. The `let` keyword is a bit shorter than `const` to write, so we only used that. This was done only for stylistic reasons.
-
-This helped our minds to not lose sight of focusing on size, but made it difficult for newcomers to start contributing to Preact. For that reason alone we switched to `prettier` and loosened our rule regarding usage of `let` or `const`. Today we use both, but you can still find many existing places where `let` is still in use.
-
-In the end there is no effect on size regardless if you use `const`, `let` or use both. Our code is downtranspiled to `ES5` for npm so both will be replaced with `var` anyways. Therefore it doesn't really matter at all which one is used in our codebase.
-
-This will only become important once shipping modern JavaScript code on npm becomes a thing and bundlers follow suit.
-
```

Comment:
I don't think we have too many weird `let`/`const`s these days, so I think this whole section is rather unnecessary? I'm not sure anyone would actually ask this question is all.

Can revert if we want it for the "Our code is downtranspiled to `ES5`...` info though.